Source/WebCore/bindings/js/JSElementCustom.cpp

```diff
@@ -148,4 +149,22 @@ JSValue JSElement::ariaOwnsElements(JSGlobalObject& lexicalGlobalObject) const
     return getElementsArrayAttribute(lexicalGlobalObject, *this, WebCore::HTMLNames::aria_ownsAttr);
 }
 
+bool JSElement::shouldEnableWebkitRequestFullScreen(ScriptExecutionContext* context)
+{
+#if ENABLE(FULLSCREEN_API)
+    RefPtr document = dynamicDowncast<Document>(context);
+    if (!document)
+        return false;
+
+    if (document->quirks().shouldDisableElementFullscreenQuirk())
+        return false;
+
+    return document->quirks().shouldEnterNativeFullscreenWhenCallingElementRequestFullscreenQuirk()
+        || document->settingsValues().fullScreenEnabled;
```

Comment:
Is there a reason this can't be
```suggestion
        || document->settings().fullScreenEnabled();
```
?

Would also be slightly better to reverse the order so that the expression that's more likely to be true is evaluated first.

Reply:
There's not a consistent reason, no. The bindings code generates the `settingsValues().<settingName>` code, but yes we could just use `document->settings()` directly.